freesound/middleware.py

```diff
@@ -77,31 +63,45 @@ def __call__(self, request):
                 cache.set(cache_key, [has_old_license, has_sounds], 2592000) # 30 days cache
                 if has_old_license and has_sounds:
                     return HttpResponseRedirect(reverse("bulk-license-change"))
-
-            else :
+            else:
                 has_old_license = cache_info[0]
                 has_sounds = cache_info[1]
-                #print "CACHE LICENSE: has_old_license=" + str(has_old_license) + " has_sounds=" + str(has_sounds)
                 if has_old_license and has_sounds:
                     return HttpResponseRedirect(reverse("bulk-license-change"))
         response = self.get_response(request)
         return response
 
 
+class FrontendPreferenceHandler(object):
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """
+        This middleware sets a session variable when the parameter 'new_frontend' is received.
+        The 'render' method will use this session variable to display the new/old frontend
+        """
+        if request.GET.get(settings.FRONTEND_CHOOSER_REQ_PARAM_NAME, None):
```

Comment:
shouldn't we at least cast this?

Reply:
Why? I mean if the param is not correctly passed we just show the default. This only needs to be passed once so I think we're fine.

Reply:
Hmm, considering that, perhaps it's complete overkill to do this in Middleware? Why not make a small view function that is `def switch_to_beastwoosh(request)` rather than the thing with parameter in any page?